File: benchmarks/poly/sortnums/sortnums.py
import sys, time


def main():
    _t0 = time.perf_counter_ns()
    n = 1000000
    seed = 12345
    arr = []
    for _ in range(n):
        seed = (48271 * seed) % 2147483647
        arr.append(seed % 1000000)

    arr.sort()

    checksum = 0
    for v in arr:
        checksum = (checksum * 31 + v) % 1000000007

    print(arr[0])
    print(arr[-1])
    print(checksum)
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: benchmarks/poly/sortnums/sortnums.go
package main

import (
	"fmt"
	"os"
	"sort"
	"time"
)

func main() {
	t0 := time.Now()
	n := 1000000
	seed := 12345
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		seed = (48271 * seed) % 2147483647
		arr[i] = seed % 1000000
	}

	sort.Ints(arr)

	checksum := 0
	for _, v := range arr {
		checksum = (checksum*31 + v) % 1000000007
	}

	fmt.Println(arr[0])
	fmt.Println(arr[n-1])
	fmt.Println(checksum)
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}
